pkg/puppetdb/facts_test.go

```diff
@@ -34,10 +34,24 @@ func TestFacts(t *testing.T) {
 	require.Equal(t, expectedFacts, actual)
 }
 
+// TestFactContents performs a test on the FactContents method, and verified the expected response is returned,
+func TestFactContents(t *testing.T) {
+	// Test with query
+	query := `[ "extract", [ "value", [ "function", "count" ] ], [ "=", "path", [ "os", "name" ] ], [ "group_by", "value" ] ]`
+	setupGetResponder(t, factContents, "query="+query, "factcontents-response.json")
+	actual, err := pdbClient.FactContents(query, nil, nil)
+	require.Nil(t, err)
+	require.Equal(t, expectedFactContents, actual)
+}
+
 var expectedFactNames = []string{"agent_canary", "agent_specified_environment", "aio_agent_build", "aio_agent_version"}
 var expectedFactPaths = []FactPath{{Path: []interface{}{"partitions", "sda3", "mount"}, Type: "string"}, {Path: []interface{}{"partitions", "sda3", "size"}, Type: "string"}, {Path: []interface{}{"partitions", "sda3", "uuid"}, Type: "string"}, {Path: []interface{}{"apt_package_dist_updates", float64(90)}, Type: "string"}}
 var expectedFacts = []Fact{
 	{Name: "id", Value: "root", Certname: "foobar.puppetlabs.net", Environment: "production"},
 	{Name: "os", Value: map[string]interface{}{"architecture": "x86_64", "distro": map[string]interface{}{"codename": "Core", "description": "CentOS Linux release 7.4.1708 (Core)", "id": "CentOS", "release": map[string]interface{}{"full": "7.4.1708", "major": "7", "minor": "4"}, "specification": ":core-4.1-amd64:core-4.1-noarch"}, "family": "RedHat", "hardware": "x86_64", "name": "CentOS", "release": map[string]interface{}{"full": "7.4.1708", "major": "7", "minor": "4"}, "selinux": map[string]interface{}{"config_mode": "permissive", "config_policy": "targeted", "current_mode": "permissive", "enabled": true, "enforced": false, "policy_version": "28"}}, Certname: "foobar.puppetlabs.net", Environment: "production"},
 	{Name: "gid", Value: "root", Certname: "foobar.puppetlabs.net", Environment: "production"},
 }
+var expectedFactContents = []Fact{
```

Comment:
Would you be better unmarshaling the file here and then that would mean that you wouldn't need to change these values in two places whenever a change is made?

Reply:
Hi @eoinmcq ... I am not sure. I copied the "TestFact" and changed it to "TestFactContents" to test the FactContents() func. I figured it would be better to make my test look like all the others than to make mine "better" ;)

Reply:
Yeah tbh I hadn't seen these tests before so I guess I am ok with that. (I think in general the codebase could maybe do that differently but not really your problem).